roles/confluent.common/tasks/main.yml

```diff
@@ -68,12 +68,12 @@
   when:
     - jolokia_enabled|bool
     - not jolokia_url_remote|bool
-    - not ansible_check_mode
 
 - name: Download Jolokia Jar
   get_url:
     url: "{{ jolokia_jar_url }}"
     dest: "{{ jolokia_jar_path }}"
+    force: true # Dest does not include version, so need to download and check if changed
```

Comment:
Can we make this a boolean parameter defaulting to false.

Reply:
> Can we make this a boolean parameter defaulting to false.

Yes, I can do that.

Reply:
I am having a hard time to find proper names for these eventual new variables. And I do not understand why we should have a variable to control if we should try to make actual state equal desired state? As it is now, there is no way to upgrade Prometheus or Jolokia. @nsharma-git can you please explain why you think a variable is better?

Reply:
IMO actual state and desired state may not be always same. We may not always want to get the latest one if there is some issue with it. Having controlled way to get the latest one would be always better.

Reply:
> We may not always want to get the latest one if there is some issue with it.

I definitely agree on that, but the issue here is that even if you change the version of Jolokia or JMX Exporter, your system will not be updated! And that is very confusing IMO. The reason being that the `get_url` module does not even attempt to do anything if the `dest` file is present. Another option would be to include the Jolokia/JMX-exporter version in the `dest` filename.

Reply:
We need to have the force flag, I'm just trying to not make it true as default value. 
We can also add some checks if file was updated when the force flag was set to true explicitly.

Reply:
@nsharma-git I have now added two new variables as requests. PTAL! 😊

Reply:
Thanks @erikgb 😊